posthog/management/commands/test_migrations_are_safe.py

```diff
@@ -67,7 +67,7 @@ def run_and_check_migration(variable):
                         sys.exit(1)
                     if "CONSTRAINT" in operation_sql and (
                         "-- existing-table-constraint-ignore" not in operation_sql
-                        or (
```

Comment:
@pauldambra I think this was meant to be an and? Otherwise constraints on new tables don't work. The conditions were [previously](https://github.com/PostHog/posthog/pull/15715/files#diff-5493309edc887361d77bd680876b201aa946fc5e27d1d70dfec063f4ef0af2f7L68) being and'd together

Reply:
all these ands/ors/nots break my brain 🤣 

When I hit this constraints were only allowed if the table was new (IIUC)

I was trying to say constraints are allowed when the table is new or when the check is explicitly ignored

Adding a foreign key (at least in the case I was changing it for) added a constraint, so the check as was stopped us adding any FK. Fair enough we might care for some tables but not for all so the ignore comment short circuits the rest of the check.

But 🤷 :)

Reply:
(totally possible I'm confused)

Reply:
Yeah i think with the and it should achieve what you intended now: table is new or ignore flag exists == table is old and ignore flag doesn't exist